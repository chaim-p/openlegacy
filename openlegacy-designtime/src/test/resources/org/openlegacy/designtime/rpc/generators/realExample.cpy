01 XXOLAKKR-PARMS.                                        
   03 LK-CUSTOMER.                                        
       05 LK-CUST-IND           PIC X.                    
       05 LK-CUST-ID            PIC 9(9).                 
       05 LK-CUST-NAME          PIC X(25).                
       05 LK-CUST-ADDRESS       PIC X(25).                
       05 LK-CUST-PHONE         PIC X(20).                
       05 LK-CARDS OCCURS 10 TIMES INDEXED BY CARD-INX.   
          07 LK-CARD-IND            PIC X.                
          07 LK-CARD-NUMBER         PIC 9(11).            
          07 LK-CARD-PREFIX         PIC 9(5).             
          07 LK-CARD-SUG-SHERUT     PIC 9(3). 