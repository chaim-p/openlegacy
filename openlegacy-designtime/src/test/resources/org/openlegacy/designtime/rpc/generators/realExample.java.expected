package test.com;

import java.math.BigInteger;
import java.util.List;

import org.openlegacy.annotations.rpc.*;
import org.openlegacy.FieldType.*;

   @RpcPart(partName="XxolakkrParms")
   public static class XxolakkrParms {

   private LkCustomer lkCustomer;
   @RpcPart(partName="LkCustomer")
   public static class LkCustomer {

   @RpcField(length =1, originalName="LK-CUST-IND")
   private String lkCustInd;

   @RpcNumericField(minimumValue=-999999999, maximumValue=999999999, decimalPlaces=0)
   @RpcField(length =9, originalName="LK-CUST-ID")
   private Integer lkCustId;

   @RpcField(length =25, originalName="LK-CUST-NAME")
   private String lkCustName;

   @RpcField(length =25, originalName="LK-CUST-ADDRESS")
   private String lkCustAddress;

   @RpcField(length =20, originalName="LK-CUST-PHONE")
   private String lkCustPhone;

   @RpcPartList(count=10)
   private List<LkCards> lkCards;
   @RpcPart(partName="LkCards")
   public static class LkCards {

   @RpcField(length =1, originalName="LK-CARD-IND")
   private String lkCardInd;

   @RpcNumericField(minimumValue=-99999999999L, maximumValue=99999999999L, decimalPlaces=0)
   @RpcField(length =11, originalName="LK-CARD-NUMBER")
   private BigInteger lkCardNumber;

   @RpcNumericField(minimumValue=-99999, maximumValue=99999, decimalPlaces=0)
   @RpcField(length =5, originalName="LK-CARD-PREFIX")
   private Integer lkCardPrefix;

   @RpcNumericField(minimumValue=-999, maximumValue=999, decimalPlaces=0)
   @RpcField(length =3, originalName="LK-CARD-SUG-SHERUT")
   private Integer lkCardSugSherut;

  }
  }
  }
